### DISCUSSION TITLE
using hpx with existing boost asio code

### DISCUSSION THREAD

**Discussion by Spongman** [STEllAR-GROUP/hpx#5615] [Q&A]
we have a rather large application that uses boost::asio and i'd like to make use of the hpx future continuations in that code, but we're running into an issue where we're unable to create hpx threads in our thread-pool implementaiton.
is there any guidance on how to do this?

**Comment by hkaiser** [comment]
We would have to go back and change all uses of Asio to Boost.Asio (i.e. headers and namespaces). This is not a principal problem, just a bit of work. This would also require some build system support.

@Spongman is that something you'd be interested in working on?

**Comment by Spongman** [comment]
so would that fix the issues creating hpx::thread? 

```
#0 0x00007f9d9e69fcd0 in hpx::threads::thread_data::get_scheduler_base (this=0x0) at /opt/hpx/include/hpx/threading_base/thread_data.hpp:511
#1 0x00007f9d9e6a0bb2 in hpx::thread::thread<void (policynet::utils::ThreadPool::*)(unsigned long), policynet::utils::ThreadPool*, unsigned long&> (this=0x7ffe52c22120, f=```

**Reply by hkaiser** [reply]
> so would that fix the issues creating hpx::thread?
> 
> ```
> #0 0x00007f9d9e69fcd0 in hpx::threads::thread_data::get_scheduler_base (this=0x0) at /opt/hpx/include/hpx/threading_base/thread_data.hpp:511
> #1 0x00007f9d9e6a0bb2 in hpx::thread::thread<void (policynet::utils::ThreadPool::*)(unsigned long), policynet::utils::ThreadPool*, unsigned long&> (this=0x7ffe52c22120, f=```
> ```

I don't know without actually seeing the code.

**Reply by msimberg** [reply]
@Spongman could you open an issue with more details (cmake configuration, ideally an example that reproduces the issue). In principle there shouldn't be any issues with HPX using standalone Asio and you using Boost.Asio and that looks like it's probably unrelated to that (`this=0x0` indicates that you may be trying to access the scheduler of an HPX thread on a non-HPX thread).
